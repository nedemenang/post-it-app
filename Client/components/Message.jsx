import React, {Component} from 'react';
import {
  BrowserRouter as Router,
  Route,
  Link
} from 'react-router-dom';
import '../public/style.css';
import $ from '../public/jquery.js';
import AppActions from '../actions/AppActions';
import {ListItem} from 'material-ui/List';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import {green100, green500, green700} from 'material-ui/styles/colors';
import Avatar from 'material-ui/Avatar';


const style = {margin: 5};
const muiTheme = getMuiTheme({
  palette: {
    primary1Color: green500,
    primary2Color: green700,
    primary3Color: green100,
  },
}, {
  avatar: {
    borderColor: null,
  },
  //userAgent: req.headers['user-agent'],
});


class Message extends Component {

  constructor(props){
    super(props);
    this.state= {
    };
  }
  render(){
    //console.log(this.props.message);
    return(
      <MuiThemeProvider muiTheme={muiTheme}>
        <ListItem leftAvatar={
        <Avatar
          src={this.props.message.userProfilePic}
          size={30}
          style={style}
        />
      }>
          <strong>{this.props.message.messageBody}</strong><br/>
          <p><small>{this.props.message.postedBy} - {this.props.message.postedon}</small></p>
        </ListItem>
      </MuiThemeProvider>
    );
  }
}

export default Message;