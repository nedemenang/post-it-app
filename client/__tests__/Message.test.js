import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Message from '../components/Message.jsx';

describe('the message item component', () => {
  let mountedComponent;
  const groupId = '-KrAwfM16qbLOig_mSOc';

  const message = {
    id: 'TUsUaucS5nb5kuOP6SpvTUcX9vD3',
    messageBody: 'Test Message',
    postedBy: 'test@postedby.com',
    postedByDisplayName: 'testUser',
    profilePic: '',
    postedon: '09/09/2017, 13:29:29',
    priority: 'normal',
    isRead: false
  };

  const userRead = [];

  const messageItem = () => {
    if (!mountedComponent) {
      mountedComponent = shallow(<Message groupId = {groupId}
      message = {message} userRead = {userRead} key = {1}/>);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('should render', () => {
    const component = messageItem();
    expect(component).toBeDefined();
  });


  it('should render without throwing an error', () => {
    const component = messageItem();
    expect(component.contains(message.messageBody)).toBe(true);
  });

  it('should match snapshot test', () => {
    const component = messageItem();
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });


  it('should recieve props', () => {
    expect(Object.keys(messageItem().props()).length).toBeGreaterThan(0);
    expect(messageItem().instance().props.message).toEqual(message);
  });
});
