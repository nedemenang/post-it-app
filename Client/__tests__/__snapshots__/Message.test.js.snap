// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the message item component should match snapshot test 1`] = `
<div>
  <ListItem
    autoGenerateNestedIndicator={true}
    containerElement="span"
    disableKeyboardFocus={false}
    disabled={false}
    initiallyOpen={false}
    insetChildren={false}
    leftAvatar={
      <Avatar
        size={30}
        src=""
        style={
          Object {
            "margin": 5,
          }
        }
      />
    }
    nestedItems={Array []}
    nestedLevel={0}
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onNestedListToggle={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    onTouchTap={[Function]}
    open={null}
    primaryTogglesNestedList={false}
    secondaryTextLines={1}
  >
    <strong>
       
      Test Message
    </strong>
    <br />
    <p>
      <small>
        test@postedby.com
         - 
      </small>
    </p>
  </ListItem>
  <Popover
    anchorOrigin={
      Object {
        "horizontal": "left",
        "vertical": "bottom",
      }
    }
    animated={true}
    autoCloseWhenOffScreen={true}
    canAutoPosition={true}
    onRequestClose={[Function]}
    open={false}
    scrollableContainer="window"
    style={
      Object {
        "overflowY": "auto",
      }
    }
    targetOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
    useLayerForClickAway={true}
    zDepth={1}
  >
    <Menu
      autoWidth={true}
      desktop={false}
      disableAutoFocus={false}
      initiallyKeyboardFocused={false}
      maxHeight={null}
      multiple={false}
      onChange={[Function]}
      onEscKeyDown={[Function]}
      onItemTouchTap={[Function]}
      onKeyDown={[Function]}
    />
  </Popover>
</div>
`;
